File: src/square.py
class Square:

    def __init__(self, a: int):
        self.a: int = a

    def area(self) -> int:
        return self.a**2

    def perimeter(self) -> int:
        return self.a * 4

def main():
    square = Square(a=2)
    print('area: ', square.area())
    print('perimeter: ', square.perimeter())

if __name__ == '__main__':
    main()

File: src/square.nim
import math, strutils

type
  Square = object
    a: int

proc area(self: Square): int =
  return self.a ^ 2

proc perimeter(self: Square): int =
  return self.a * 4

proc main() =
  let square = Square(a: 2)
  echo "area: ", square.area()
  echo "perimeter: ", square.perimeter()

when isMainModule:
  main()
